rust/content.md

```diff
@@ -33,20 +33,20 @@ $ docker run -it --rm --name my-running-app my-rust-app
 This creates an image that has all of the rust tooling for the image, which is 1.8gb. If you just want the compiled application:
 
 ```dockerfile
-FROM rust:1.40 as builder
+FROM %%IMAGE%%:1.40 as builder
 WORKDIR /usr/src/myapp
 COPY . .
 RUN cargo install --path .
 
 FROM debian:buster-slim
-RUN apt-get update && apt-get install -y extra-runtime-dependencies && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get install -y <extra-runtime-dependencies> && rm -rf /var/lib/apt/lists/*
```

Comment:
Minor nitpick is that `<` and `>` are shell syntax (minus a missing place to put the output) and users will get: `Syntax error: "&&" unexpected`. I think that error is less helpful than the current `Unable to locate package extra-runtime-dependencies`. So I'd recommend either leaving it as is or changing it to something that doesn't look similar to a package name like "put-your-runtime-packages-here", while still adding the expanded `Note` on line 47.

Using square brackets instead isn't a great option since `apt-get` attempts to interpret it as a glob and then regex to find packages to install.

---

CC rust image maintainer, @sfackler. What do you want to do here (on this line and with the PR)?

Reply:
if someone publish `extra-runtime-dependencies` or `put-your-runtime-packages-here` package with malicious code, thing cloud get ugly here. 🤔

Reply:
Just to clarify, this malicious "someone" would need to be a Debian Developer (https://wiki.debian.org/DebianDeveloper/JoinTheProject), _and_ they'd have to get it past the Debian `NEW` process (https://wiki.debian.org/NewQueue), _and_ they'd have to convince the Debian Release Team (https://wiki.debian.org/Teams/ReleaseTeam) to let it into a stable release, so it's a reasonably high bar with a lot of separate independent review steps where such malicious intent would be very likely to be caught before hitting very many (if any) users.

(Alternatively, I guess they could hack the Debian infrastructure to serve their malicious package, but they'd have to ensure it's all signed correctly _and_ mirrored, and go unnoticed long enough for users to get the packages.  If someone goes to all that trouble, I doubt publishing a _new_ malicious `extra-runtime-dependencies` or `put-your-runtime-packages-here` package is going to be the top of their list of things to attack.)

Reply:
I just tried installing `extra-runtime-dependencies` and ending up here. Happy 2023, guys!

Reply:
@vaniusrb Happy new year!

Reply:
I think using all caps would probably be easier.

```Dockerfile
RUN apt-get update && apt-get install -y EXTRA_RUNTIME_DEPENDENCIES_HERE && rm -rf /var/lib/apt/lists/*
```

Or maybe even better, have is a comment

```Dockerfile
FROM %%IMAGE%%:1.40 as builder
WORKDIR /usr/src/myapp
COPY . .
RUN cargo install --path .

FROM debian:buster-slim

# Use this section if you need extra dependencies
# RUN apt-get update \
#     && apt-get install -y \
#         <extra-runtime-dependencies> \
#     && rm -rf /var/lib/apt/lists/*

COPY --from=builder /usr/local/cargo/bin/myapp /usr/local/bin/myapp
CMD ["myapp"]
```